Desktop/widgets/factorsformbase.cpp

```diff
@@ -47,9 +48,7 @@ void FactorsFormBase::setUpModel()
 	_initNumberFactors = property("initNumberFactors").toInt();
 
 	QQuickItem::connect(this, SIGNAL(titleChanged(int, QString)), _factorsModel, SLOT(titleChangedSlot(int, QString)));
-	QQuickItem::connect(this, SIGNAL(factorAdded(int, QVariant)), _factorsModel, SLOT(factorAddedSlot(int, QVariant)));
-
-	connect(_factorsModel, &ListModelFactorsForm::addListView, this, &FactorsFormBase::addListViewSlot);
+	QQuickItem::connect(this, SIGNAL(factorAdded(int, QVariant)), this, SLOT(factorsAddedSlot(int, QVariant)));
```

Comment:
Why again with the MACROs? Can't you refer to the actual function-definitions like we do everywhere?

Reply:
factorAdded & titleChanged are QML signals

Reply:
But it still comes from `factorsformbase` right? You are referring to `this` after all.
You could just make a `Q_INVOKABLE void factorAdded(int QVariant);` signal in `factorsformbase.h` .
The same for `titleChanged`.

That way the signal definition is in C++ and you can use a normal connect that does typechecks etc.